src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -302,9 +317,30 @@ public function configureOptions(OptionsResolver $resolver)
             return $choiceTranslationDomain;
         };
 
+        $multipleNormalizer = function (Options $options, $multiple) {
+            if (in_array($options['widget'], array('radio', 'checkbox'), true)) {
+                return 'checkbox' === $options['widget'];
+            }
+
+            return $multiple;
+        };
+
+        $expandedNomalizer = function (Options $options, $expanded) {
+            if (null !== $expanded) {
+                @trigger_error('The "expanded" option is deprecated since version 3.1 and will be removed in 4.0. You should use "widget" instead.', E_USER_DEPRECATED);
+            } else {
+                $expanded = false;
```

Comment:
This line is not need if you cast return to bool :

`return (bool) (in_array($options['widget'], array('radio', 'checkbox'), true) ?: $expanded);`

anyway the cast would be done anywhere in this class as it's only a test variable. Then it has to remain internal and not to be removed in 4.0 with the actual implementation IMO.